openedx/core/djangoapps/content/search/models.py

```diff
@@ -0,0 +1,56 @@
+"""Database models for content search"""
+
+from __future__ import annotations
+
+from typing import List
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from opaque_keys.edx.django.models import LearningContextKeyField
+from rest_framework.request import Request
+
+from cms.djangoapps.contentstore.views.course import get_courses_accessible_to_user
+from openedx.core.djangoapps.content_libraries.api import get_libraries_for_user
+
+
+class SearchAccess(models.Model):
```

Comment:
1. Should this model be named something more general, e.g. `LearningContext`?
2. Should it live somewhere more generally accessible, e.g. one level up at `openedx.core.djangoapps.content`?

cf [relevant comments](https://github.com/openedx/modular-learning/issues/198#issuecomment-2043131883)

Reply:
@ormsbee @bradenmacdonald ^

Reply:
I'd just as soon as leave it here. I think there is a place for a more general model, but that's probably an ADR and some careful thinking since it's going to be used by _all the things_. So I'd rather have something more modestly scoped here for now.